### PR TITLE
Update scenario tests for the design v2

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[4012](https://github.com/ballerina-platform/ballerina-standard-library/issues/4012)

## Checklist
- [x] Linked to an issue~
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests

### CODE REVIEW

File: examples/medical-center/Ballerina.toml [hunk 1]
```diff
@@ -2,7 +2,14 @@
...
 
 [build-options]
 observabilityIncluded = false
+
+[persist.entity.storage.mysql]
+host = "localhost"
+port = 3308
+user = "root"
+password = "Test123#"
+database = "medical_center_db"
```
Comment (daneshk) [inline examples/medical-center/Ballerina.toml:15]:
```suggestion
database = "medical_center_db"

```

File: examples/medical-center/Config.toml [hunk 2]
```diff
@@ -1,6 +1,8 @@
-[medical_center]
+[medical_center.entity]
```
Comment (daneshk) [inline examples/medical-center/Config.toml:1]:
```suggestion
[medical_center.entities]
```

File: examples/medical-center/persist/entity.bal [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+|};
+
+type MedicalNeed record {|
+    readonly int needId;
+    int itemId;
+    int beneficiaryId;
+    time:Civil period;
+    string urgency;
+    int quantity;
+|};
```
Comment (daneshk) [inline examples/medical-center/persist/entity.bal:18]:
```suggestion
|};

```

File: examples/rainier/persist/rainier.bal [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    string notes;
+|};
+
+type Workspace record {|
+    readonly string workspaceId;
+    string workspaceType;
+
+    Building location;
+    Employee employee;
+|};
```
Comment (daneshk) [inline examples/rainier/persist/rainier.bal:63]:
```suggestion
|};

```

**Comment by daneshk** [comment]
@sahanHe https://github.com/ballerina-platform/persist-tools/blob/main/settings.gradle#L24 please uncomment this

File: examples/medical-center/main.bal [hunk 5]
```diff
@@ -1,6 +1,6 @@
-// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline examples/medical-center/main.bal:1]: Please undo this license header change

File: examples/medical-center/main.bal [hunk 6]
```diff
@@ -15,102 +15,111 @@
...
         name: "item name",
-        'type: "type1",
+        itemType: "type1",
         unit: "ml"
     };
-    MedicalItem createdItem = check miClient->create(item);
-    io:println("Created item id: ", createdItem.itemId);
-
-    MedicalItem retrievedItem = check miClient->readByKey(1);
+    io:println("1");
```
Comment (daneshk) [inline examples/medical-center/main.bal:28]: shall we remove these debug prints

File: examples/medical-center/main.bal [hunk 7]
```diff
@@ -15,102 +15,111 @@
...
+    int[] itemIds = check mcClient->/medicalitem.post([item]);
+    io:println("Created item id: ", itemIds[0]);
+    io:println("1");
+    entity:MedicalItem retrievedItem = check mcClient->/medicalitem/[itemIds[0]].get();
     io:println("Retrieved item: ", retrievedItem);
 
-    MedicalItem|error itemError = miClient->readByKey(20);
+    entity:MedicalItem|error itemError = mcClient->/medicalitem/[5].get();
```
Comment (daneshk) [inline examples/medical-center/main.bal:35]:
```suggestion
    entity:MedicalItem|error itemError = mcClient->/medicalitem/[5]();
```

Comment (daneshk) [reply examples/medical-center/main.bal:35]: no need to get, since it is the default function

File: examples/rainier/main.bal [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+    string[] buildingCodes = check rainierClient->/building.post([building1]);
+    rainier:Building buildingRetrieved = check rainierClient->/building/[building1.buildingCode].get();
+    buildingCodes = check rainierClient->/building.post([building2, building3]);
+    buildingRetrieved = check rainierClient->/building/[building2.buildingCode].get();
+    buildingRetrieved = check rainierClient->/building/[building3.buildingCode].get();
+    buildingRetrieved = check rainierClient->/building/[building1.buildingCode].get();
+    stream<rainier:Building, error?> buildingStream = rainierClient->/building.get();
+    rainier:Building[] buildings = check from rainier:Building building_temp in buildingStream
+        select building_temp;
+    buildingRetrieved = check rainierClient->/building/[building1.buildingCode].put({
```
Comment (daneshk) [inline examples/rainier/main.bal:189]:
```suggestion
    buildingUpdated = check rainierClient->/building/[building1.buildingCode].put({
```

File: examples/rainier/main.bal [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+
+    rainier:Building|error buildingRetrievedError = rainierClient->/building/["invalid-building-code"].put({
+        city: "Galle",
+        state: "Southern Province",
+        postalCode: "10890"
+    });
+    if buildingRetrievedError !is persist:Error {
+        panic error("Error expected");
+    }
+    buildingRetrieved = check rainierClient->/building/[building1.buildingCode].delete();
```
Comment (daneshk) [inline examples/rainier/main.bal:205]:
```suggestion
    buildingDeleted = check rainierClient->/building/[building1.buildingCode].delete();
```

Comment (sahanHe) [reply examples/rainier/main.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/medical-center/main.bal (hunk 7 above)
Comment (sahanHe) [reply examples/medical-center/main.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/medical-center/main.bal (hunk 6 above)
Comment (sahanHe) [reply examples/medical-center/main.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/medical-center/main.bal (hunk 5 above)
Comment (sahanHe) [reply examples/medical-center/main.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/rainier/persist/rainier.bal (hunk 4 above)
Comment (sahanHe) [reply examples/rainier/persist/rainier.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/medical-center/persist/entity.bal (hunk 3 above)
Comment (sahanHe) [reply examples/medical-center/persist/entity.bal]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

File: examples/medical-center/Ballerina.toml (hunk 1 above)
Comment (sahanHe) [reply examples/medical-center/Ballerina.toml]: updated in [29c72ce](https://github.com/ballerina-platform/persist-tools/pull/120/commits/29c72ce16d2edf4882433055270167e57360d0f7)

**Review by daneshk — APPROVED** [review]

**Pull request merged**